### PR TITLE
Clean up fake mounters.

### LINKED ISSUE #61502: Clean up fake mounters
**Is this a BUG REPORT or FEATURE REQUEST?**:
@kubernetes/sig-storage-bugs 
/kind cleanup

**What happened**:
There are too many FakeMounter implementations in the codebase.  It's annoying to have to update all of them whenever we have to change the Mounter interface.  They should all be consolidated to use the FakeMounter implementation in pkg/util/mount/fake.go.

### PR DESCRIPTION
**Pull request by cofyc**
**What this PR does / why we need it**:

Fixes https://github.com/kubernetes/kubernetes/issues/61502

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

list of fake mounters:

- (keep) pkg/util/mount.FakeMounter
- (removed) pkg/kubelet/cm.fakeMountInterface:
- (inherit from mount.FakeMounter) pkg/util/mount.fakeMounter
- (inherit from mount.FakeMounter) pkg/util/removeall.fakeMounter
- (removed) pkg/volume/host_path.fakeFileTypeChecker

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cofyc** [comment]
@feiskyer Thanks!

/assign @msau42 @jsafrane

**Comment by jsafrane** [comment]
@cofyc, please apply for membership using this process, it will simplify things for you. I can sponsor you, perhaps @msau42 can too. https://github.com/kubernetes/community/blob/master/community-membership.md#member

**Comment by cofyc** [comment]
@jsafrane Thanks! I have applied for membership a few days ago. I'm waiting for response.

**Comment by msau42** [comment]
/lgtm
Thanks!

**Comment by cofyc** [comment]
@brendandburns @dchen1107 @lavalamp @smarterclayton @thockin
hi, Could you help approve this?

**Comment by cofyc** [comment]
/assign @brendandburns @dchen1107 @lavalamp @smarterclayton @thockin
hi, Could you help approve this?

File: pkg/util/mount/exec_mount_test.go [hunk 1]
```diff
@@ -94,85 +92,26 @@ func TestUnmount(t *testing.T) {
...
 	mounter := NewExecMounter(exec, wrappedMounter)
 
 	mounter.Unmount(destinationPath)
 }
 
 /* Fake wrapped mounter */
 type fakeMounter struct {
-	t *testing.T
+	*FakeMounter
+	*testing.T
```
Comment (lavalamp) [inline pkg/util/mount/exec_mount_test.go]: keep the name lowercase?

File: pkg/util/mount/fake.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ import (
 type FakeMounter struct {
 	MountPoints []MountPoint
 	Log         []FakeAction
+	Filesystem  map[string]FileType
```
Comment (thockin) [inline pkg/util/mount/fake.go]: This type should be moved to a testing/ subdir so it doesn't get linked into non-test code.  Won't block this PR on it.

File: pkg/util/mount/exec_mount_test.go (hunk 1 above)
Comment (cofyc) [reply pkg/util/mount/exec_mount_test.go]: Done. `T` changed to `t`. (`fakeMounter` needs to embed `FakeMounter` to inherit methods so it cannot be lowered)

File: pkg/util/mount/fake.go (hunk 2 above)
Comment (cofyc) [reply pkg/util/mount/fake.go]: OK. I'll change a lot of go imports and BUILD files, I'll create another PR for this.

**Comment by cofyc** [comment]
@jsafrane @msau42 

I've made a minor change, please help re-lgtm it, thanks!

- `fakeMounter.T` changed to `fakeMounter.t` in `pkg/util/mount/exec_mount_test.go`

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**